packages/core/src/internal/io.js

```diff
@@ -114,18 +114,11 @@ export function spawn(fn, ...args) {
   return detach(fork(fn, ...args))
 }
 
-const printJoinDeprecationWarning = once(() =>
-  log('warn', '`join(...tasks)` is deprecated. Please use `join([...tasks])` to join multiple tasks.'),
-)
-
 export function join(taskOrTasks) {
-  if (arguments.length > 1) {
-    if (process.env.NODE_ENV === 'development') {
-      printJoinDeprecationWarning()
-    }
-    taskOrTasks = array.from(arguments)
-  }
   if (process.env.NODE_ENV === 'development') {
+    if (arguments.length > 1) {
+      throw new Error('join(...tasks) is not supported any more. Please use join([...tasks]) to join multiple tasks.')
```

Comment:
`Please use join(tasks)`?

Reply:
Should I rephrase this error message?

Reply:
Hm, not sure which one is better - `join([...tasks])` seems to indicate that the argument type is an array, `join(tasks)` does not hold any information regarding this (except pluralized word)

Reply:
that's not a big deal. we can keep it as is. I meant, probably it could be confusing - for example, why we require to make a copy of array before passing to effect?

Reply:
let's keep it then, I doubt people will actually make those copies, at least not most commonly